### PR TITLE
Don't fail pipeline init if github token secret exists

### PR DESCRIPTION
**Pull request by SoManyHs**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by SoManyHs** [comment]
now, when secret exists:
```
~/workspace/aws-reinvent-2018-trivia-game(master)$ archer pipeline init
? Would you like to add an environment to your pipeline? Yes
? Which environment would you like to add to your pipeline? david
? Would you like to add an environment to your pipeline? No
? What is your application's GitHub repository? https://github.com/SoManyHs/aws-reinvent-2019-trivia-game
? Please enter your GitHub Personal Access Token for your repository: https://github.com/SoManyHs/aws-reinvent-2019-trivia-game [? for help] ****
✔ Success! Secret already exists for https://github.com/SoManyHs/aws-reinvent-2019-trivia-game! Do nothing.
✔ Success! Wrote the pipeline manifest for https://github.com/SoManyHs/aws-reinvent-2019-trivia-game at 'ecs-project/pipeline.yml'
✔ Success! Wrote the buildspec for the pipeline's build stage at 'ecs-project/buildspec.yml'
The manifest contains configurations for your CodePipeline resources, such as your pipeline stages and build steps.
The buildspec contains the commands to build and push your container images to your ECR repositories.

Recommended follow-up actions:
- Update the build phase of your buildspec to unit test your applications before pushing the images.
- Update your pipeline manifest to add additional stages.
- Run `archer pipeline update` to deploy your pipeline for the repository.

~/workspace/aws-reinvent-2018-trivia-game(master)$ cat ecs-project/pipeline.yml 
# This YAML file defines the relationship and deployment ordering of your environments.

# The name of the pipeline
name: pipeline-badgoose-aws-reinvent-2019-trivia-game

# The version of the schema used in this template
version: 1

# This section defines the source artifacts.
source:
  # The name of the provider that is used to store the source artifacts.
  provider: GitHub
  # Additional properties that further specifies the exact location
  # the artifacts should be sourced from. For example, the GitHub provider
  # has the following properties: repository, branch.
  properties:
    access_token_secret: github-token-badgoose-aws-reinvent-2019-trivia-game
    branch: master
    repository: https://github.com/SoManyHs/aws-reinvent-2019-trivia-game

# The deployment section defines the order the pipeline will deploy
# to your environments.
stages:
    -
      # The name of the environment to deploy to.
      name: david


```

File: internal/pkg/cli/pipeline_init.go [hunk 1]
```diff
@@ -119,10 +119,18 @@ func (opts *InitPipelineOpts) Validate() error {
 func (opts *InitPipelineOpts) Execute() error {
 	secretName := opts.createSecretName()
 	secretArn, err := opts.secretsmanager.CreateSecret(secretName, opts.GitHubAccessToken)
+
 	if err != nil {
-		return err
+		// TODO don't fail if secret exists
```
Comment (efekarakus) [inline internal/pkg/cli/pipeline_init.go]: should we remove the TODO now :)

File: internal/pkg/cli/pipeline_init.go [hunk 2]
```diff
@@ -119,10 +119,18 @@ func (opts *InitPipelineOpts) Validate() error {
...
+		var existsErr *secretsmanager.ErrSecretAlreadyExists
+		if errors.As(err, &existsErr) {
+			// Do nothing if the secret already exists.
+			log.Successf("Secret already exists for %s! Do nothing.\n", color.HighlightUserInput(opts.GitHubRepo))
+		} else {
```
Comment (efekarakus) [inline internal/pkg/cli/pipeline_init.go]: nit: by flipping the if-statement we can get rid of the else clause.
```go
if !errors.As(err, &existsErr) {
   return err
}
log.Success("hurray")
```

File: internal/pkg/cli/pipeline_init.go [hunk 3]
```diff
@@ -119,10 +119,18 @@ func (opts *InitPipelineOpts) Validate() error {
...
+			return err
+		}
 	}
-	opts.secretArn = secretArn
+	opts.secretArn = secretArn // Maybe don't need?
```
Comment (efekarakus) [inline internal/pkg/cli/pipeline_init.go]: yeah let's remove it 👍 we can add it if there is a need later

**Comment by sonofachamp** [comment]
Could there be a scenario where the secret name might already exist, but the value needs to be updated?

File: internal/pkg/cli/pipeline_init.go [hunk 4]
```diff
@@ -118,11 +117,15 @@ func (opts *InitPipelineOpts) Validate() error {
 // Execute writes the pipeline manifest file.
 func (opts *InitPipelineOpts) Execute() error {
 	secretName := opts.createSecretName()
-	secretArn, err := opts.secretsmanager.CreateSecret(secretName, opts.GitHubAccessToken)
+	_, err := opts.secretsmanager.CreateSecret(secretName, opts.GitHubAccessToken)
```
Comment (sonofachamp) [inline internal/pkg/cli/pipeline_init.go]: Do we not need to capture the secret ARN on successful creates any more?

Comment (SoManyHs) [reply internal/pkg/cli/pipeline_init.go]: we never used it -- see https://github.com/aws/amazon-ecs-cli-v2/pull/390#discussion_r347111774

**Pull request merged**